pkg/grpc/client.go

```diff
@@ -0,0 +1,76 @@
+package grpc
+
+// gRPC client implementation to be able to connect to the gRPC server and perform the following operations:
+// - Start a remote JupyterLab server
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/grpc/jupyter"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	requestTimeout = 30 * time.Second
+)
+
+type GrpcClient struct {
+	conn          *grpc.ClientConn
+	token         string
+	jupyterClient jupyter.JupyterServerHostClient
+}
+
+func New() *GrpcClient {
+	return &GrpcClient{}
+}
+
+// Connects to the gRPC server on the given port
+func (g *GrpcClient) Connect(ctx context.Context, port int, token string) error {
+	// Attempt to connect to the given port
+	conn, err := grpc.Dial(fmt.Sprintf("localhost:%d", port), grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+
+	if err != nil {
+		return fmt.Errorf("Failed to connect to the internal server on port %d", port)
+	}
+
+	g.conn = conn
+	g.token = token
+	g.jupyterClient = jupyter.NewJupyterServerHostClient(conn)
```

Comment:
nit: this would be better named: `jupyter.NewClient` or `jupyter.NewServerHostClient`

Reply:
`jupyter.JupyterServerHostClient` is auto-generated and is based on the name of the service defined in the proto. These are already in place in the server and Codespaces extension and would require updates to both of those as well if we wanted to change the service name.